modules/market/src/main/java/com/opengamma/strata/market/interpolator/InterpolatorCurveExtrapolator.java

```diff
@@ -69,46 +66,42 @@ public String toString() {
    * Bound extrapolator.
    */
   static class Bound implements BoundCurveExtrapolator {
-    private final Interpolator1D interpolator;
-    private final Interpolator1DDataBundle dataBundle;
+    private AbstractBoundCurveInterpolator interpolator;
```

Comment:
This needs to be `final`.

(In Java, `final` variables have a particular status wrt the Java Memory Model which affects immutability and thread-safety)